### ISSUE TITLE
Streaming issue

### ISSUE THREAD

**Issue by kcarnold** [rytido/ncf-tech#11]
We had what seemed to be sync issues with the stream on 2021-04-25.

I have reported this on the BlackMagic forums. No reply yet. 

I am suspicious about a recent firmware update applied on the ATEM. I propose we revert to 8.6.0 or earlier, especially if we don't get a chance to test it.

Known problems happened on:

* 2021-04-25
* 2021-06-06
* 2021-06-27

Details of 2021-06-27 incident:

* The only two issues I noticed when re-watching the sermon (admittedly, at 2x...) were:
    * https://youtu.be/_dW7jIroK9A?t=2849 (47m29)
    * https://youtu.be/_dW7jIroK9A?t=3623 (“There is a crooked path… that leads us a___” = 56:02 — “May God in Christ” = 57:45 in the recording) -- so it lost 103 seconds of video here. But Aaron had only brought the ATEM off-air for 5 seconds. (So where did all that video go??)
* But Ryan (watching remotely) reports: "It was maybe 3 times for a few seconds each when I first sent a text, and 2 times a little bit later"

**Comment by kcarnold** [comment]
Investigation continues. I have not tried downgrading the firmware since it felt more risky, but that may be the next course of action.

Fortunately this is rare, but unfortunately that means that debugging it is very tricky.

We have noticed that sometimes when the error occurs YouTube gives an error: "YouTube is not receiving enough video to maintain smooth streaming."

One wild theory is that one of the clocks on the ATEM is prone to drifting, causing it to include timecode data in the RTMP stream that was inconsistent with the number of frames that it had sent, which caused YouTube to think that the stream was lagging (so giving the error message) or leading (so causing it to drop what it thought was "old" video in order to catch up to live). Though this doesn't completely make sense, one thing that does lead me to suspect this is that the OBS recording has a lot of duplicated frames (i.e., two consecutive frames are identical). Even though OBS is running at 30 fps and the ATEM is currently at 60fps, if the clocks are consistent it should at least be getting a new frame each time. So that leads me to wonder if some clock is actually drifting around... but it would have to be a *lot* of drift to explain that.

**Comment by kcarnold** [comment]
Update: this seems to be Internet-related after all. We had several drops this morning, enough to actually lose viewers. But earlier, Proclaim was not downloading slides at first, then later fixed itself. And during one of the outages, Proclaim also showed a "reconnecting" message or similar. So it seems highly likely that this is related, and possibly likely that it's a Comcast problem.

Fortunately I'd had the whole setup going through an nginx-rtmp mirror on the Proclaim Mac, so we had a pristine recording, which I trimmed and uploaded.

Setting YouTube to add 30sec extra delay seemed to help. I upped it to 1 minute this evening.

I'm thinking of adding a cloud server with something like https://github.com/sistason/rtmp_stream_delayer to give an extra big buffer.
